rspace/src/main/scala/coop/rchain/rspace/history/RadixTree.scala

```diff
@@ -201,7 +214,13 @@ object RadixTree {
           decodeItem(pos0Next, nodeNext) // Try to decode next item.
         }
 
-      decodeItem(0, emptyNode)
+      try {
+        decodeItem(0, emptyNode)
+      } catch {
+        case _: Exception =>
+          assert(assertion = false, "Error during deserialization: invalid data format")
+          emptyNode
+      }
```

Comment:
This code confuses me, but I don't know how to write it correctly. In the current codebase, I did not find the use of try-catch. Why use assert instead of exception? Maybe consider using an Option, Either, or an Attempt? @tgrospic, what do you think?

Reply:
If we cannot raise Applicative error (@hilltracer have measured slowdown) we can use suppress warning annotation which will allow the use of `throw` so that original exception can be wrapped.

```scala
@SuppressWarnings(Array("org.wartremover.warts.Throw"))
def decode(bv: ByteVector): Node = {
  ...
  try {
    decodeItem(0, emptyNode)
  } catch {
    case cause: Exception =>
      throw new Exception("Error during deserialization: invalid data format", cause)
  }
}
```